presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -806,6 +806,13 @@ protected Scope visitJoin(Join node, Scope scope)
 
                 addCoercionForJoinCriteria(node, leftExpression, rightExpression);
                 expressions.add(new ComparisonExpression(EQUAL, leftExpression, rightExpression));
+
+                if (node.getType() == Join.Type.LEFT) {
+                    output = createScope(node, scope, left.getRelationType().joinWith(right.hide(rightExpression).getRelationType()));
```

Comment:
I don't think this is the right semantic. The fields are not supposed to be "hidden". They should both be replaced by a single column with no relation alias.